docs/lab/06/index.md

```diff
@@ -0,0 +1,359 @@
+:::
+
+#### Transmission example
+
+1. main sets `CS` to `LOW` - the sub that the main wants to communicate with is activated - all other subs disconnect from `MOSI` and `MISO` lines
+2. main writes the first bit on the `MOSI` line, and, *simultaneously*, sub writes the first bit on the `MISO` line
+3. main starts the clock
+4. on *rising edge* 
+    - main reads data from the `MISO` line 
+    - sub reads data from the `MOSI` line
+5. on *falling edge*
+    - main writes data to the `MOSI` line
+    - sub writes data to the `MISO` line
+6. repeat steps 4 and 5 until main stops the clock
+
+:::note
+Whether steps 4 and 5 happen on rising/falling edge depends on the configuration of the SPI.
+:::
+
+![SPI_example](images/spi_network_transmission.svg)
+
+In the image above, the main first exchages data with the first sub by setting its corresponding `CS` line to low. After it finishes transmission with sub 1, it deactivates it and, afterwards, initiates communication with the second sub by setting its `CS` line to low. Notice that both main and sub read and write to the MOSI/MISO line simultaneously.
+
+### SPI modes
+
+SPI has 4 different modes which define when data is read or written. These modes can be configured through 2 registers: `CPOL` and `CPHA`.
+
+| Mode | `CPOL` | `CPHA` |
+|-|-|-|
+| 0 | 0 | 0 |
+| 1 | 0 | 1 |
+| 2 | 1 | 0 |
+| 3 | 1 | 1 |
+
+| | | |
+|-|-|-| 
+| `CPOL` | Clock polarity | defines when the data bit is read <br/> 0: *rising edge* <br/> 1: *falling edge* |
+| `CPHA` | Clock phase | defines when the data is written to the line <br/> 0: when `CS` *activates* <br/> 1: on *clock edge* (depends on `CPOL`) |
+
+### Daisy Chaining
+
+An SPI network with many sub devices can prove impractical from a hardware point of view. **Daisy chaining** is a method that solves this by linking multiple devices together with the following configuration:
+
+![SPI_Daisy](images/spi_daisy.svg)
+
+The `MOSI` line connects the main device only to the first sub. Then, the `MISO` output of the first sub ties to the `MOSI` input of the second sub, and so forth. The final sub in the chain ties to the `MISO` line to the main device. All `CS` lines are common, therefore all subs are active at the same time.
+
+If main wants to communicate with the second sub, it will send its data through the first sub, and receive data from the second sub through all subs following it. (TODO: explain)
+
+## DMA
+
+When it comes to standard communication protocols that rely on [MMIO](/docs/lab/02), it would be inefficient to rely on the MCU itself to handle all of these data transfers. This is why the **Direct Memory Access** (DMA) is used. Its purpose is to offload the MCU by dealing with simple transmission to and from peripherals, and whenever it finishes a transfer, it raises an interrupt.
+
+## SPI in Embassy
+
+In Embassy, we can use the SPI both blocking and asynchronously. The following example will be using the asynchronous version.
```

Comment:
Add a warning explaining why the synchronous versions is not good.

Reply:
Done.